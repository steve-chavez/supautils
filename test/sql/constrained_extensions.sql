-- this feature is only enabled for pg >= 13, for pg12 this will be false
SELECT current_setting('server_version_num')::int >= 130000 as gte_pg13;

-- constrained by cpu
create extension adminpack;
\echo

-- constrained by memory
create extension cube;
\echo

-- constrained by disk
create extension lo;
\echo

-- passes all resource constraints
create extension amcheck;
\echo

-- no resource constraints
create extension bloom;
