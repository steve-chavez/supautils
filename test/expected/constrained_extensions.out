-- this feature is only enabled for pg >= 13, for pg12 this will be false
SELECT current_setting('server_version_num')::int >= 130000 as gte_pg13;
 gte_pg13 
----------
 t
(1 row)

-- constrained by cpu
create extension adminpack;
ERROR:  not enough CPUs for using this extension
DETAIL:  required CPUs: 64
HINT:  upgrade to an instance with higher resources
\echo

-- constrained by memory
create extension cube;
ERROR:  not enough memory for using this extension
DETAIL:  required memory: 17 GB
HINT:  upgrade to an instance with higher resources
\echo

-- constrained by disk
create extension lo;
ERROR:  not enough free disk space for using this extension
DETAIL:  required free disk space: 20 GB
HINT:  upgrade to an instance with higher resources
\echo

-- passes all resource constraints
create extension amcheck;
\echo

-- no resource constraints
create extension bloom;
